pkg/issuer/vault/setup.go

```diff
@@ -45,6 +45,7 @@ const (
 	messageAppRoleAuthFieldsRequired = "Vault AppRole auth requires both roleId and tokenSecretRef.name"
 )
 
+// Setup validates the vault by setting IssuerConditions, logging and errs in case there are errors on the client setup.
```

Comment:
Suggestion:
```suggestion
// Setup creates a new Vault client and attempts to authenticate with the Vault instance and sets the issuer's conditions to reflect the success of the setup.
```
or something like that if you agree